File: apps/ios-app/EcommerceMobile/Config/AppConfig.swift
import Foundation

enum AppConfig {
    private static let info = Bundle.main.infoDictionary ?? [:]

    static let defaultBaseURLString = info["APIBaseURL"] as? String ?? "https://api.example.com"
    static let defaultTenantId = info["DefaultTenantId"] as? String ?? "public"
    static let defaultUsername = info["DefaultUsername"] as? String ?? ""
    static let defaultScope = "api1 offline_access"
    static let defaultOidcScope = "openid profile api1 offline_access"
    static let mobileRedirectURIString = "dsmobile://auth/callback"

    static var allowPasswordGrantFallback: Bool {
        info["AllowPasswordGrantFallback"] as? Bool ?? false
    }

    static var baseURL: URL {
        guard let url = URL(string: defaultBaseURLString) else {
            preconditionFailure("Invalid API base URL: \(defaultBaseURLString)")
        }
#if !DEBUG
        guard url.scheme == "https", url.host != "api.example.com" else {
            preconditionFailure("Release builds must configure a real https API base URL.")
        }
#endif
        return url
    }

    static var mobileRedirectURL: URL {
        guard let url = URL(string: mobileRedirectURIString) else {
            preconditionFailure("Invalid mobile redirect URI: \(mobileRedirectURIString)")
        }
        return url
    }
}

File: apps/android-app/app/src/main/java/com/ds/ecommerce/mobile/data/AppConfig.kt
package com.ds.ecommerce.mobile.data

import com.ds.ecommerce.mobile.BuildConfig

object AppConfig {
    const val apiBaseUrl: String = BuildConfig.API_BASE_URL
    const val defaultTenantId: String = BuildConfig.DEFAULT_TENANT_ID
    const val defaultUsername: String = BuildConfig.DEFAULT_USERNAME
    const val defaultScope: String = "api1 offline_access"
    const val defaultOidcScope: String = BuildConfig.MOBILE_OIDC_SCOPE
    const val mobileRedirectUri: String = BuildConfig.MOBILE_REDIRECT_URI
    const val allowPasswordGrantFallback: Boolean = BuildConfig.ALLOW_PASSWORD_GRANT_FALLBACK

    init {
        if (!BuildConfig.DEBUG) {
            require(apiBaseUrl.startsWith("https://")) {
                "Release builds must point to an https mobile API base URL."
            }
            require(!apiBaseUrl.contains("example.com")) {
                "Release builds must replace the placeholder mobile API base URL."
            }
        }
    }
}
